addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java

```diff
@@ -1137,4 +1242,67 @@ public void handlePushCommand(String command, String payload) {
                 }
         }
     }
+
+    public void updateNotifications(ZonedDateTime currentTime, ZonedDateTime now,
+            @Nullable JsonCommandPayloadPushNotificationChange pushPayload, JsonNotificationResponse[] notifications) {
+        Device device = this.device;
+        if (device == null) {
+            return;
+        }
+
+        ZonedDateTime nextReminder = null;
+        ZonedDateTime nextAlarm = null;
+        ZonedDateTime nextMusicAlarm = null;
+        ZonedDateTime nextTimer = null;
+        for (JsonNotificationResponse notification : notifications) {
+            if (StringUtils.equals(notification.deviceSerialNumber, device.serialNumber)) {
+                // notification for this device
+                if (StringUtils.equals(notification.status, "ON")) {
+                    if (StringUtils.equals(notification.type, "Reminder")) {
+                        String offset = ZoneId.systemDefault().getRules().getOffset(Instant.now()).toString();
+                        ZonedDateTime alarmTime = ZonedDateTime
+                                .parse(notification.originalDate + "T" + notification.originalTime + offset);
+                        if (StringUtils.isNotBlank(notification.recurringPattern) && alarmTime.isBefore(now)) {
+                            continue; // Ignore recurring entry if alarm time is before now
+                        }
+                        if (nextReminder == null || alarmTime.isBefore(nextReminder)) {
+                            nextReminder = alarmTime;
+                        }
+                    } else if (StringUtils.equals(notification.type, "Timer")) {
```

Comment:
```suggestion
                    } else if ("Timer".equals(notification.type)) {
```
I don't think you need `StringUtils` here. If you put the constant at the left side it will guard against null pointers.

Reply:
Changed, but I'am not sure if it is a good way. I know that it makes technical the same, but the semantic is different. I want to say, check if the variable is something, and not something is the same as the variable.

Reply:
I guess it's a matter of what your used to. I must say I've never interpreted it the way you describe it and always read it as being symmetric without any meaning on which is on which side. That doesn't mean your interpretation is wrong, just different.
In code quality guidelines the approach I recommend here is also recommended as a way of defensive programming. In general when core methods like equals can be used it's preferred over using third party libraries.